src/MCPClient/lib/clientScripts/archivematicaCreateMETS2.py

```diff
@@ -827,7 +829,66 @@ def createFileSec(directoryPath, parentDiv, baseDirectoryPath, baseDirectoryName
             file_elem.set("DMDID", dspaceMetsDMDID)
     
     return structMapDiv
-        
+
+def build_arranged_structmap(original_structmap, fileGroupIdentifier):
+    """
+    Given a structMap, builds a new copy of the structMap with file and directory labels assigned according to their intellectual arrangement.
+    Logical arrangement is determined using the levels of description which were assigned to them during SIP arrange.
+
+    :param etree.Element original_structmap: the structMap on which the arranged structMap should be based.
+    :param str fileGroupIdentifier: the SIP's UUID
```

Comment:
Especially since this is new code, maybe we can start referring to sip_uuid instead of the ever-confusing fileGroupIdentifier?